docs/tutorial/Tutorial.md

```diff
@@ -0,0 +1,501 @@
+   8. In `post_copy` add the following code. Also, add an inline (always) attribute.
+       ```rust
+       forwarding_word::clear_forwarding_bits::<VM>(obj);
+       ```
+5. Add a new public structure, `MyGCProcessEdges`, with the type parameter `<VM:VMBinding>`. It will hold an instance of `ProcessEdgesBase` and `PhantomData`, and implement the Default trait:
+    ```rust
+    #[derive(Default)]
+    pub struct MyGCProcessEdges<VM: VMBinding> {
+        base: ProcessEdgesBase<MyGCProcessEdges<VM>>,
+        phantom: PhantomData<VM>,
+    }
+    ```
+6. Add a new implementations block `impl<VM:VMBinding> ProcessEdgesWork for MyGCProcessEdges<VM>`.
+   1. Add a VM type alias (`type VM = VM`).
+   2. Add a new method, `new`.
+       ```rust
+       fn new(edges: Vec<Address>, _roots: bool) -> Self {
+           Self {
+               base: ProcessEdgesBase::new(edges),
+               ..Default::default()
+           }
+       }
+      ```
+   3. Add a new method, `trace_object(&mut self, object: ObjectReference)`.
+     1. This method should return an ObjectReference, and use the inline (*not* always) attribute.
+     2. Check if the object passed into the function is null (`object.is_null()`). If it is, return the object.
+     3. Check if the object is in the tospace (`self.plan().tospace().in_space(object)`). If it is, call `trace_object` through the tospace to check if the object is alive, and return the result:
+         ```rust
+         self.plan().tospace().trace_object(
+               self,
+               object,
+               super::global::ALLOC_MyGC,
+               self.worker().local(),
+           )
+         ```
+     4. If it is not in the tospace, check if the object is in the fromspace and return the result of the fromspace's `trace_object` if it is.
+     5. If it is in neither space, it must be in the immortal space, or large object space. Trace the object with `self.plan().common.trace_object(self, object)`.
+7. Add two new implementation blocks, `Deref` and `DerefMut` for `MyGCProcessEdges`. 
+    ```rust
+    impl<VM: VMBinding> Deref for MyGCProcessEdges<VM> {
+        type Target = ProcessEdgesBase<Self>;
+        #[inline]
+        fn deref(&self) -> &Self::Target {
+            &self.base
+        }
+    }
+
+    impl<VM: VMBinding> DerefMut for MyGCProcessEdges<VM> {
+        #[inline]
+        fn deref_mut(&mut self) -> &mut Self::Target {
+            &mut self.base
+        }
+    }
+    ```
+8. A few import statements need to be added to the other files so that they can use the functions in `gc_works`.
+   1. `global.rs`: Import `MyGCCopyContext` and `MyGCProcessEdges`.
+   2. `mod.rs`: Import `gc_works` as a module (`mod gc_works;`).
+   
+5. In `global.rs`, delete `handle_user_collection_request`. This function was an override of a Common plan function, which can run correctly when collection is handled.   
+
```

Comment:
I believe we need to add `schedule_collection()` to `MyGC`, like here https://github.com/mmtk/mmtk-core/blob/66dbdce4e3b392c260b9f4601764b53379877eed/src/plan/semispace/global.rs#L93